htlcswitch/circuit_map.go

```diff
@@ -280,15 +283,45 @@ func (cm *circuitMap) restoreMemState() error {
 
 			// Retrieve the pending circuit, set its keystone, then
 			// add it to the opened map.
-			circuit := pending[inKey]
-			circuit.Outgoing = outKey
-			opened[*outKey] = circuit
+			circuit, ok := pending[inKey]
+			if ok {
+				circuit.Outgoing = outKey
+				opened[*outKey] = circuit
+			} else {
+				strayKeystones = append(strayKeystones, Keystone{
+					InKey:  inKey,
+					OutKey: *outKey,
+				})
+			}
 
 			return nil
 		}); err != nil {
 			return err
 		}
 
+		// If any stray keystones were found, we'll proceed to prune
+		// them from the circuit map's persistent storage. This may
+		// manifest on older nodes that had updated channels before
+		// their short channel id was set properly. We believe this
+		// issue has been fixed, though this will allow older nodes to
+		// recover without additional intervention.
+		for _, strayKeystone := range strayKeystones {
```

Comment:
Are we sure that these issues are only affecting locally initiated payments?

Reply:
Or is there a greater issue here, in that we actually don't properly ACK back settles/fails by locally initiated payments?

Reply:
this is purely a result of allowing pending channels to begin forwarding htlcs too soon, meaning the outgoing chanid is 0:0:0 when this should never have been allowed in the first place

Reply:
only seen one person with this issue in the wild